src/compiler/watch.ts

```diff
@@ -194,23 +194,105 @@ namespace ts {
         };
     }
 
+    export const enum WatchType {
+        ConfigFile = "Config file",
+        SourceFile = "Source file",
+        MissingFile = "Missing file",
+        WildcardDirectory = "Wild card directory",
+        FailedLookupLocations = "Failed Lookup Locations",
+        TypeRoots = "Type roots"
+    }
+
+    interface WatchFactory<X, Y= undefined> extends ts.WatchFactory<X, Y> {
+        watchLogLevel: WatchLogLevel;
+        writeLog: (s: string) => void;
+    }
+
+    export function createWatchFactory<Y = undefined>(host: { trace?(s: string): void; }, options: { extendedDiagnostics?: boolean; diagnostics?: boolean; }) {
+        const watchLogLevel = host.trace ? options.extendedDiagnostics ? WatchLogLevel.Verbose :

```

Comment:
this formatting is misleading since the inner-wrapped flat series of conditionals has a line-break in the middle. Is there a way to make the whole conditional a flat series of checks? That would be easier to read than the nested `host.trace ? options.extendedDiagnostics ? ...` style.